src/components/categories-filter/CategoriesFilter.component.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+
+import appStyles from '@/App.module.css';
+import { CategoriesDataProviderComponent } from '@/data-providers/CategoriesDataProvider.component.tsx';
+import { transformCheckedCategories, transformCheckedCategoryIds } from '@/helpers/categoriesFilterHelper.ts';
+import type { CheckedCategoriesProps } from '@/types/CheckedCategoriesProps.type.tsx';
+
+import styles from './categories-filter.module.css';
+
+interface CategoriesFilterProps {
+    categoryIds?: number[];
+    onCategoriesChange: (categoryIds: number[]) => void;
+}
+
+export const CategoriesFilterComponent: React.FC<CategoriesFilterProps> = ({ categoryIds, onCategoriesChange }) => {
+    const [checkedCategories, setCheckedCategories] = useState<CheckedCategoriesProps>(() => transformCheckedCategories(categoryIds || []));
```

Comment:
`Props` we add to React component props, here it confused

```suggestion
    const [checkedCategories, setCheckedCategories] = useState<CheckedCategories>(() => transformCheckedCategories(categoryIds || []));
```

Reply:
fixed